### ISSUE TITLE
Support for IPv6 (AAAA) records in DNS-based peer discovery

### ISSUE THREAD

**Issue by jleroy** [rabbitmq/rabbitmq-server#1171]
RabbitMQ does not work on an IPv6-only network:
```
rabbitmq-server[28777]: ERROR: epmd error for host XXXX.XXXX.XXXX: nxdomain (non-existing domain)
```
The problem seems to be located in https://github.com/rabbitmq/rabbitmq-server/blob/master/src/rabbit_peer_discovery_dns.erl#L70. The discover_hostnames() function does not check for AAAA record.

**Comment by michaelklishin** [comment]
@jleroy your findings are not correct. First of all, the module you are looking at is not a part of any GA release yet, it will only be available in 3.7.0. "nxdomain" comes from epmd, which is not a part of RabbitMQ but rather an Erlang/OTP daemon, and [it does support IPv6](https://github.com/erlang/otp/pull/864/commits/14dab9bf7b125269f84b3a0799698932fa12c9c3).

There can be things in RabbitMQ that assume IPv4 but we aren't aware of them except for the (unreleased) module above.

**Comment by michaelklishin** [comment]
Peer discovery DNS backend (currently in master only) now [supports both IPv6/AAAA and IPv4/A records](https://github.com/rabbitmq/rabbitmq-server/pull/1189) thanks to @Gsantomaggio.
